pgwatch2/metrics/db_stats/14/metric.sql

```diff
@@ -0,0 +1,35 @@
+select
+  (extract(epoch from now()) * 1e9)::int8 as epoch_ns,
+  numbackends,
+  xact_commit,
+  xact_rollback,
+  blks_read,
+  blks_hit,
+  tup_returned,
+  tup_fetched,
+  tup_inserted,
+  tup_updated,
+  tup_deleted,
+  conflicts,
+  temp_files,
+  temp_bytes,
+  deadlocks,
+  blk_read_time,
+  blk_write_time,
+  extract(epoch from (now() - pg_postmaster_start_time()))::int8 as postmaster_uptime_s,
+  extract(epoch from (now() - pg_backup_start_time()))::int8 as backup_duration_s,
+  checksum_failures,
+  extract(epoch from (now() - checksum_last_failure))::int8 as checksum_last_failure_s,
+  case when pg_is_in_recovery() then 1 else 0 end as in_recovery_int,
+  system_identifier::text as tag_sys_id,
+  session_time::int8,
+  active_time::int8,
+  idle_in_transaction_time::int8,
```

Comment:
Why are these cast to int8 ? They should, like most other things in pg_stat_database, already be bigint, so no conversion should actually happen.
Since none of the other columns that where previously already bigint are cast to int8, I don't think we need it here?

Reply:
@laurenz since you wrote those commits, can you please share your thoughts on this?

Reply:
Right, don't worry about overflow.
Internally, these values are stored in *microseconds* as `int64` (= `bigint`):
```c
typedef int64 PgStat_Counter;

[...]

typedef struct PgStat_StatDBEntry
{
    Oid         databaseid;
    [...]
    PgStat_Counter total_session_time;
    PgStat_Counter total_active_time;
    PgStat_Counter total_idle_in_xact_time;
    [...]
}
```

So in PostgreSQL they would overflow after 29 years.

The values are divided by 1000 and output as `double precision` upon display.

So neither using `bigint` not using `double precision` would be a problem.
But why not use `double precision`? You'd lose nothing and might gain some decimal places. And math on `double precision` is very fast too.

BTW, the whole feature is under danger of getting reverted: See [this thread](https://postgr.es/m/20210801205501.nyxzxoelqoo4x2qc%40alap3.anarazel.de) on the hackers list.

Reply:
29 years...so you wouldn't really be able to send a spaceship to distant galaxies if it ran on PostgreSQL then :thinking:  :laughing: 

Thank you for enlightening us!
I'm not sure about `bigint` vs. `double precision` either. While the latter's precision advantage is surely quite quickly irrelevant after a short while, it might be useful for differential views, e.g. when wanting to assess how much time a was spent over a certain time frame - though the aggragation intervals that we're dealing with in pgwatch2 are usually a few minutes to a few hours. And I'm not sure if I need to know decimal places of a millisecond value if I am aggregating over as few as five minutes.

So maybe we should hold out a bit to see if that thread goes somewhere that nullifies these counters? @kmoppel-cognite
I'd say let's give it one more week...